File: LXUtilities/UIKitExtensions/WKWebView+LXExtension.m
//
//  WKWebView+LXExtension.m
//
//  Created by 从今以后 on 16/10/10.
//  Copyright © 2016年 从今以后. All rights reserved.
//

#import "WKWebView+LXExtension.h"

@implementation WKWebView (LXExtension)

- (void)lx_loadRequestWithURLString:(NSString *)URLString
{
    [self loadRequest:[NSURLRequest requestWithURL:[NSURL URLWithString:URLString]]];
}

- (UIImage *)lx_snapshot
{
    UIView *contentView = [self.scrollView valueForKey:@"_scrollNotificationViews"];
    UIGraphicsBeginImageContextWithOptions(contentView.bounds.size, NO, 0);
    [contentView.layer renderInContext:UIGraphicsGetCurrentContext()];
    UIImage *snapshot = UIGraphicsGetImageFromCurrentImageContext();
    UIGraphicsEndImageContext();
    return snapshot;
}

@end

File: LXUtilities/UIKitExtensions/Swift/WKWebView+LXExtension.swift
//
//  WKWebView+LXExtension.swift
//
//  Created by 从今以后 on 2017/11/9.
//  Copyright © 2017年 从今以后. All rights reserved.
//

import WebKit

extension Swifty where Base: WKWebView {
	
	func loadRequest(withURLString string: String) throws {
		guard let url = URL(string: string) else {
			throw URLError(.badURL)
		}
		base.load(URLRequest(url: url))
	}
}
